### PR TITLE
Add check-ses-limits python script

### PR DESCRIPTION
**Pull request by jdmulloy**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] Have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?

### CODE REVIEW

**Review by hchen15 — APPROVED** [review]

**Comment by feanil** [comment]
What's the value of leaving the number of thresholds open-ended?  Why not just take a warn threshold and a critical threshold explicitly?

**Comment by jdmulloy** [comment]
@feanil Probably not much value if we won't be wanting to add more thresholds. I could change it to take a warn and crit threshold via options. If I do that should I have defaults or require them to be provided?

File: util/jenkins/check-ses-limits.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# Copied from https://stackoverflow.com/a/41153081
+class ExtendAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest) or []
+        items.extend(values)
+        setattr(namespace, self.dest, items)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-c', '--crit', required=True, type=float,
```
Comment (feanil) [inline util/jenkins/check-ses-limits.py]: If we're going to have a word let's just make it the whole word `--critical`

File: util/jenkins/check-ses-limits.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    help="Critical threshold in percentage")
+parser.add_argument('-w', '--warn', required=False, type=float,
+                    help="Warning threshold in percentage (Optional)")
+parser.add_argument('-r', '--region', dest='regions', nargs='+',
+                    action=ExtendAction, required=True,
+                    help="AWS regions to check")
+args = parser.parse_args()
+
+if args.warn and args.warn >= args.crit:
+    print("ERROR: Warning threshold (" + str(args.warn) +
```
Comment (feanil) [inline util/jenkins/check-ses-limits.py]: Use, `str.format` instead of string concatenation.

File: util/jenkins/check-ses-limits.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+exit_code = 0
+
+session = boto3.session.Session()
+for region in args.regions:
+    ses = session.client('ses', region_name=region)
+    data = ses.get_send_quota()
+    limit = data["Max24HourSend"]
+    current = data["SentLast24Hours"]
+    percent = current/limit
+    out_str = str(current) + "/" + str(limit) + " (" + str(percent) + "%)"
```
Comment (feanil) [inline util/jenkins/check-ses-limits.py]: Use `str.format` instead of string concatenation.

File: util/jenkins/check-ses-limits.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    if percent >= args.crit:
+        print(region + " " + out_str + " - CRITICAL!")
```
Comment (feanil) [inline util/jenkins/check-ses-limits.py]: Use `str.format` instead of string concatenation.

File: util/jenkins/check-ses-limits.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        exit_code += 1
+    elif args.warn and percent >= args.warn:
+        print(region + " " + out_str + " - WARNING!")
```
Comment (feanil) [inline util/jenkins/check-ses-limits.py]: Use `str.format` instead of string concatenation.

File: util/jenkins/check-ses-limits.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+# Copied from https://stackoverflow.com/a/41153081
+class ExtendAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest) or []
+        items.extend(values)
+        setattr(namespace, self.dest, items)
+
+
+parser = argparse.ArgumentParser()
```
Comment (feanil) [inline util/jenkins/check-ses-limits.py]: Put all of the logic inside a `if __name__ == "__main__":` block.
http://ibiblio.org/g2swap/byteofpython/read/module-name.html

**Review by feanil — CHANGES_REQUESTED** [review]

**Review by feanil — APPROVED** [review]

**Pull request merged**